### DISCUSSION TITLE
How to extract mesh exterior and interior boundaries?

### DISCUSSION THREAD

**Discussion by ryan-kipawa** [DHI/mikeio#379] [Q&A]
Hi,

I have a big mesh where I would like to extract the outer and inner boundaries to shapely polygons. The "to_shapely" function seems to take a long time for the mesh size (well over 10 million elements). I also don't need each element as a polygon, just the boundaries. Is there a more efficient way to do this? 

Edit: 
I tried merging together polygons from mesh.boundary_polylines.exteriors. It's not working as I expected though. I also tried with interiors, however, some of the polylines have only 2 nodes and can't be converted to a polygon.

`polygons = [Polygon(p.xy) for p in msh.boundary_polylines.exteriors]`
`boundary = MultiPolygon(polygons)`

![image](https://user-images.githubusercontent.com/74630349/175975102-7ac239b5-bb23-459f-bcf2-d892fbac42af.png)

Edit 2:
It looks like some of the exterior polylines are misread as interiors (see image). Could it be a bug or a corrupt mesh file?

![image](https://user-images.githubusercontent.com/74630349/176005959-8d67a920-12e2-451f-a8c5-2645bafe9741.png)

Edit 3:
I found this that seems to be a workaround. A similar solution works for the interiors.

```
from shapely.ops import linemerge

lines = [LineString(l.xy) for l in msh.geometry.boundary_polylines.exteriors]
lines = [*lines, *[LineString(l.xy) for l in msh.geometry.boundary_polylines.interiors]]
lines = MultiLineString(lines)
lines = linemerge(lines)

outerLine = lines[0]
for line in lines:
    if line.length > outerLine.length:
        outerLine = line
outer_boundary = Polygon(outerLine)
```

Thanks
Ryan

**Comment by jsmariegaard** [comment]
Hi Ryan - great that you found a workaround - Unfortunately, I do not have time to look into this as I am on my way to summer vacation.
